duke-core/src/main/java/no/priv/garshol/duke/cleaners/StopwordsCleaner.java

```diff
@@ -0,0 +1,72 @@
+
+package no.priv.garshol.duke.cleaners;
+
+import no.priv.garshol.duke.Cleaner;
+
+import java.io.*;
+import java.util.List;
+import java.util.ArrayList;
+
+/**
+ * A cleaner which removes english stopwords from a string.
+ */
+
+public class StopwordsCleaner implements Cleaner {
+    private LowerCaseNormalizeCleaner sub;
+    private String[] stopwords;
+    private ArrayList<String> wordsList = new ArrayList<String>();
+
+
+    public StopwordsCleaner() {
+        this.sub = new LowerCaseNormalizeCleaner();
+
+        try {
+            this.stopwords = loadStopwords();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+
+    public String clean(String value) {
+
+        value = sub.clean(value);
+        if (value == null || value.equals(""))
+            return value;
+
+
+        String[] words = value.split(" ");
+        for (String word : words) {
+            wordsList.add(word);
+        }
+
+        for (int j = 0; j < stopwords.length; j++) {
+            if (wordsList.contains(stopwords[j])) {
+                wordsList.remove(stopwords[j]);
```

Comment:
This is also inefficient, because `.remove` requires the rest of the array to be copied downwards. So performance will be O(n^2). I'd propose merging the two for loops. Basically, turn the top loop into

```
for (String word : words) {
  if (!stopwords.contains(word))
    wordsList.add(word);
}
```

This should be not just simpler but also much more efficient.